server/historian/packages/historian-base/src/routes/utils.ts

```diff
@@ -118,18 +123,36 @@ export async function createGitService(createArgs: createGitServiceArgs): Promis
 	const customData: ITenantCustomDataExternal = details.customData;
 	const writeToExternalStorage = !!customData?.externalStorageData;
 	const storageUrl = config.get("storageUrl") as string | undefined;
+	const ignoreEphemeralFlag: boolean = config.get("ignoreEphemeralFlag");
 	const maxCacheableSummarySize: number =
 		config.get("restGitService:maxCacheableSummarySize") ?? 1_000_000_000; // default: 1gb
 
-	let isEphemeral = cache ? isEphemeralContainer : false;
+	let isEphemeral: boolean = isEphemeralContainer;
 	if (!ignoreEphemeralFlag) {
-		if (isEphemeralContainer !== undefined) {
-			await cache?.set(`isEphemeral:${documentId}`, isEphemeralContainer);
+		const isEphemeralKey: string = `isEphemeralContainer:${documentId}`;
+		if (isEphemeral !== undefined && isEphemeral !== null) {
+			// If an isEphemeral flag was passed in, cache it in Redis
+			Lumberjack.info(`Setting ${isEphemeralKey} to ${isEphemeral}`);
+			await cache?.set(isEphemeralKey, isEphemeral);
 		} else {
-			isEphemeral = await cache?.get(`isEphemeral:${documentId}`);
-			// Todo: If isEphemeral is still undefined fetch the value from database
+			isEphemeral = await cache?.get(isEphemeralKey);
+			if (isEphemeral === null || isEphemeral === undefined) {
+				// If isEphemeral is still null/undefined fetch the value from database
+				try {
+					const staticProps: IDocumentStaticProperties =
+						await documentManager.readStaticProperties(tenantId, documentId);
+					isEphemeral = staticProps?.isEphemeralContainer ?? false;
+					await cache?.set(isEphemeralKey, isEphemeral); // Cache the value from the static data
+				} catch (e) {
+					isEphemeral = false;
+				}
+			}
 		}
+	} else {
+		isEphemeral = false;
 	}
+	Lumberjack.info(`Document ${documentId} is ephemeral? ${isEphemeral}`);
```

Comment:
can we pass in the document id and tenant id as the param to lumberjack.info instead of as a part of the message itself. That way, they end up become cols in kusto which is useful for filtering logs